### PR TITLE
Dashboard alerts page

### LINKED ISSUE #30: dashboard: add train alert page
### Task
- [x] add a page to the dashboard so users can receive alerts if an operator is affected by an incident.

### User Story
As a user, it would be valuable to know if a train line I usually use may be unavailable to me as it could affect my journey.

### PR DESCRIPTION
**Pull request by fm1psy**
Closes issue #30 

The incidents_alerts page now lets users give their details and subscribe to an operator

### CODE REVIEW

File: dashboard/pages/incident_alerts.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            aws_access_key_id=environ['AWS_ACCESS_KEY'],
+            aws_secret_access_key=environ['AWS_SECRET_KEY'],
+        )
+    except Exception as e:  # pylint: disable=broad-exception-caught
+        st.write(e)
+        return None
+
+
+def does_topic_arn_exist(client, topic_arn: str) -> bool:
+    """"""
```
Comment (FarihaChoudhury) [inline dashboard/pages/incident_alerts.py]: I think you forgot these docstrings

Comment (FarihaChoudhury) [reply dashboard/pages/incident_alerts.py]: and the ones in the functions below

File: dashboard/pages/daily_report.py
Comment (A-Faris) [inline dashboard/pages/daily_report.py:1]: I'm just assuming your auto-formatter did these changes

File: dashboard/pages/incident_alerts.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            if not valid_phone:
+                st.error("Invalid phone number")
+
+        return None
+
+
+def deploy_alerts_page():
+    """This serves as the main code for the alerts page"""
+    st.title("Train Alerts")
+    st.write("""Subscribe to an Operator Alert using your phone number,
```
Comment (A-Faris) [inline dashboard/pages/incident_alerts.py]:
```suggestion
    st.write("""Subscribe to an Operator Alert using your phone number,
```

Comment (A-Faris) [reply dashboard/pages/incident_alerts.py]: remove trailing whitespace

File: dashboard/pages/incident_alerts.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+def subscribe_to_topic(client, topic_arn: str, protocol: str, endpoint: str) -> None:
+    """"""
+    try:
+        return client.subscribe(TopicArn=topic_arn, Protocol=protocol, Endpoint=endpoint)
+    except:
+        return None
+
+
+def get_list_of_operators() -> list[str]:
+    """"""
```
Comment (A-Faris) [inline dashboard/pages/incident_alerts.py]: Docstrings empty

File: dashboard/pages/incident_alerts.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+
+def does_topic_arn_exist(client, topic_arn: str) -> bool:
+    """"""
+    try:
+        response = client.get_topic_attributes(
+            TopicArn=topic_arn
+        )
+        return response is not None
+    except:
```
Comment (A-Faris) [inline dashboard/pages/incident_alerts.py]:
```suggestion
    except Exception:  # pylint: disable=broad-exception-caught
```

File: dashboard/pages/incident_alerts.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    client.create_topic(
+        Name=f"c11-trainwreck-{operator_code}"
+    )
+
+
+def subscribe_to_topic(client, topic_arn: str, protocol: str, endpoint: str) -> None:
+    """"""
+    try:
+        return client.subscribe(TopicArn=topic_arn, Protocol=protocol, Endpoint=endpoint)
+    except:
```
Comment (A-Faris) [inline dashboard/pages/incident_alerts.py]:
```suggestion
    except Exception:  # pylint: disable=broad-exception-caught
```

**Review by A-Faris — CHANGES_REQUESTED** [review]
Looks good just needs pylinting

File: dashboard/pages/daily_report.py
Comment (fm1psy) [reply dashboard/pages/daily_report.py:1]: Looks like that is the case yeah

**Review by A-Faris — DISMISSED** [review]
Looks great

**Review by A-Faris — DISMISSED** [review]
Looks good

**Review by FarihaChoudhury — DISMISSED** [review]

**Review by FarihaChoudhury — DISMISSED** [review]
beautiful

**Review by Zhi-704 — APPROVED** [review]

**Pull request merged**